test/files/run/repl-classbased.check

```diff
@@ -6,18 +6,26 @@ scala> class C { implicit val k: K = K("OK?"); override def toString = s"C($k)"
 defined class C
 
 scala> val c = new C
-c: C = C(K(OK?))
+c: INSTANCE.C = C(K(OK?))
 
 scala> import c.k
 import c.k
 
 scala> implicitly[K]
-res0: K = K(OK?)
+res0: INSTANCE.K = K(OK?)
 
 scala> val k = 42
+<console>:13: warning: imported `k' is permanently hidden by definition of value k
+import c.k
```

Comment:
Seems like we've removed the special case in the typechecker that used to suppress this warning.